File: leet_0392_subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if s == '':
            return True
        ls = len(s)
        lt = len(t)
        i, j = 0, 0
        while i < ls and j < lt:
            if s[i] == t[j]:
                i += 1
            if i == ls:
                return True
            j += 1
        return False

File: leet_0392_subsequence.cc
class Solution {
public:
    bool isSubsequence(string s, string t)
    {
        if (s == "")
            return true;
        int S = s.length();
        int T = t.length();
        int i = 0, j = 0;
        while (i < S && j < T)
        {
            if (s[i] == t[j])
            {
                ++i;
            }
            ++j;
            if (i == S)
                return true;
        }
        return false;
    }
};
